src/Raven.Client/Documents/Session/EntityToBlittable.cs

```diff
@@ -55,6 +55,9 @@ public BlittableJsonReaderObject ConvertEntityToBlittable(object entity, Documen
             DocumentInfo documentInfo,
             bool removeIdentityProperty = true)
         {
+            if (value == null)
```

Comment:
Is this required? If no, then we should remove this to avoid adding this

Reply:
I think it is, won't `ConvertEntityToBlittableInternal` throw on null?

Reply:
It is required only for CompareExchange as its value can be null. Renamed the function name to ConvertToBlittableForCompareExchangeIfNeeded.